File: Submissions/#0568_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
 
  vector<int> abc(3);
  cin >> abc[0] >> abc[1] >> abc[2];

  int d;
  cin >> d;

  sort(abc.begin(), abc.end());
  
  int ab = abc[0] + d - abc[1];
  int bc = abc[0] + d * 2 - abc[2];

  if (max(ab, bc) < 0) {
    cout << 0 << endl;
  } else {
    cout << max(ab, bc) << endl;
  }

}

File: Submissions/#0568_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int[] abc = new int[3];
    abc[0] = sc.nextInt();
    abc[1] = sc.nextInt();
    abc[2] = sc.nextInt();
    int d = sc.nextInt();
    sc.close();

    Arrays.sort(abc);

    int ab = abc[0] + d - abc[1];
    int bc = abc[0] + d * 2 - abc[2];

    if (Math.max(ab, bc) < 0) {
      System.out.println(0);
    } else {
      System.out.println(Math.max(ab, bc));
    }

  }


}